lib/providers/product.dart

```diff
@@ -17,9 +21,28 @@ class Product with ChangeNotifier {
     this.isFavorite = false,
   });
 
-  ///If the favorite button is pressed, the item will be added as favorite
-  void toggleFavoriteStatus() {
-    isFavorite = !isFavorite;
+  void _setFavValue(bool newValue) {
+    isFavorite = newValue;
     notifyListeners();
   }
+
+  ///If the favorite button is pressed, the item will be added as favorite
+  Future<void> toggleFavoriteStatus() async {
+    final oldStatus = isFavorite;
+    final url = '${Constants.url}/products/$id.json';
+    try {
+      final response = await http.patch(
+        url,
+        body: json.encode({'isFavorite': !isFavorite}),
+      );
+      if (response.statusCode >= 400) {
+        _setFavValue(oldStatus);
+      } else {
+        isFavorite = !isFavorite;
+        notifyListeners();
```

Comment:
this seem DRY since you have a method where you set the new value for `isFavorite`